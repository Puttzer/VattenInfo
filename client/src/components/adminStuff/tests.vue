<template>
  <div class="page-contaier">
    <v-row class="d-flex flex-column">
      <v-row class="d-flex flex-row">
        <h1>Tester</h1>
        <v-spacer></v-spacer>
        <v-btn class="mr-12 mt-6" @click="showAddTest = !showAddTest"
          >LÃ¤gg till <br />
          Test</v-btn
        >
        <AddTest v-show="showAddTest === true" />
      </v-row>
      <v-col class="d-flex flex-row">
        <v-card
          v-for="test in this.tests.tests"
          :key="test.id"
          width="200px"
          height="250px"
          class="ma-2"
        >
          <v-list class="d-flex flex-column justify-center">
            <v-img
              :src="test.image"
              height="150px"
              width="150px"
              name="testimage"
            ></v-img>
            <v-divider></v-divider>
            <v-list-title class="Heading-2" name="testName">
              Title : {{ test.testname }}
            </v-list-title>
            <v-list-subtitle name="testShort">
              Description :{{ test.short_description }}
            </v-list-subtitle>
            <v-list-content name="testLong">
              Detailjer :{{ test.long_description }}
            </v-list-content>
            <v-list-title name="testPrice">
              Pris :{{ test.price }} SEK
            </v-list-title>
            <div class="d-flex flex-row">
              <v-icon @click="editTest(test._id)">mode_edit</v-icon>
              <v-icon @click="deleteTest(test._id)">delete</v-icon>
            </div>
          </v-list>
        </v-card>
      </v-col>
    </v-row>
  </div>
</template>

<script>
import { mapState } from "vuex";
import AddTest from "./AddTest.vue";
export default {
  data() {
    return {
      showAddTest: false,
    };
  },
  name: "tests",
  components: {
    AddTest,
  },
  computed: {
    ...mapState(["tests"]),
  },
  mounted() {
    this.$store.dispatch("tests/getTests");
  },
  methods: {
    deleteTest(_id) {
      this.$store.dispatch("tests/deleteTest", _id);
    },
  },
};
</script>

<style scoped>
.dontshow {
  display: none;
}
</style>