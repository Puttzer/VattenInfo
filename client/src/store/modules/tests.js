import Vue from 'vue'
export default {
    state: {
        tests: [],
        test: {},
        count: 0,
        selectedTests: [],
        showSelectedTests: false,
        totalAmount: 0,
        paginatedTests: [],
        numberOfPages: null

    },
    getters: {


    },
    actions: {
        async createNewTest({ commit }, formData) {
            console.log(formData)
            const token = localStorage.getItem('token')
            const response = await fetch('${process.env.VUE_APP_BACKEND_URL}/api/test/create', {
                method: 'POST',
                body: formData,
                headers: {
                    'authorization': token,
                }
            });
            const data = await response.json();
            console.log(data)
            commit('INSERT_TEST', data.newTest, { module: 'tests' })
            // commit('UPDATE_ISLOGGEDIN', false, { module: 'admin' });
        },
        async updateTest({ commit }, { formData, id }) {
            console.log('move to actions');
            console.log(formData)
            console.log(id)
            const token = localStorage.getItem('token')
            const response = await fetch(`${process.env.VUE_APP_BACKEND_URL}/api/test/update/${id}`, {
                method: 'PUT',
                body: formData,
                headers: {
                    'authorization': token,
                }
            });
            const data = await response.json();
            console.log(data)
            commit('EDIT_TEST', data.updatedTest, { module: 'tests' })
            // commit('UPDATE_ISLOGGEDIN', false, { module: 'admin' });
        },
        async getTestInfo({ commit }, id) {

            const response = await fetch(`${process.env.VUE_APP_BACKEND_URL}/api/test/${id}`, {
                method: 'GET',
                headers: {
                    'Content-Type': 'application/json',
                }
            });
            const data = await response.json();
            console.log(data)
            commit('UPDATE_TEST', data.test, { module: 'tests' })
            // commit('UPDATE_ISLOGGEDIN', false, { module: 'admin' });
        },
        async getTests({ commit }) {
            const token = localStorage.getItem('token')
            const response = await fetch('${process.env.VUE_APP_BACKEND_URL}/api/tests', {
                method: 'GET',
                headers: {
                    'Content-Type': 'application/json',
                    'authorization': token,
                }
            });
            const data = await response.json();
            // console.log(data)
            commit('updateTest', data.tests, { module: 'tests' })
            // commit('UPDATE_ISLOGGEDIN', false, { module: 'admin' });


        },
        async getPaginatedTests({ commit }, pageNumber) {
            console.log('get paginated tests')
            const token = localStorage.getItem('token')
            const page = pageNumber;
            const limit = 10;
            const response = await fetch(`${process.env.VUE_APP_BACKEND_URL}/api/paginated/tests?page=${page}&limit=${limit}`, {
                method: 'GET',
                headers: {
                    'Content-Type': 'application/json',
                    'authorization': token,
                }
            });
            const data = await response.json();
            console.log(data)
            commit('UPDATE_PAGINATED_TESTS', data.tests.results, { module: 'tests' })
            commit('UPDATE_PAGINATED_PAGES', data.tests.pages, { module: 'tests' })
            // commit('UPDATE_ISLOGGEDIN', false, { module: 'admin' });


        },
        async deleteTest({ commit }, _id) {
            const token = localStorage.getItem('token')
            const response = await fetch(`${process.env.VUE_APP_BACKEND_URL}/api/test/${_id}`, {
                method: 'DELETE',
                headers: {
                    'Content-Type': 'application/json',
                    'authorization': token,
                }
            });
            const data = await response.json()
            console.log(data)
            commit('DELETE_TEST', data.test._id, { module: 'tests' })
        }


    },
    mutations: {
        updateTest(state, value) {
            state.tests = value.map((test) => test)
            // console.log(state.tests);
        },
        DELETE_TEST(state, value) {
            console.log(state)
            const findTest = state.tests.find(test => test._id === value)
            const remainingTests = state.tests.filter((test) => {
                test._id != value
            })
            state.tests = remainingTests
            Vue.$vToastify.success(`Användaren ${findTest.testname} är bortatagen`)
            // console.log(state.tests)

        },
        INSERT_TEST(state, value) {
            state.tests.push(value)
            Vue.$vToastify.success(`användaren  ${value.testname} är  tillagd i datbasen`)

        },
        EDIT_TEST(state, { updatedTestInfo, _id }) {
            console.log(_id, updatedTestInfo)
            const findTest = state.tests.find(test => test._id === _id)
            console.log(findTest)
            findTest.testname = updatedTestInfo.testname
            findTest.testtype = updatedTestInfo.testtype
            findTest.category = updatedTestInfo.category
            findTest.price = updatedTestInfo.price
            findTest.description = updatedTestInfo.description
            findTest._id = _id
            findTest.image = updatedTestInfo.image

            state.tests.filter(test => test._id === _id).push(findTest)
            Vue.$vToastify.success(` ${findTest.testname} är uppdaterad`)

        },
        UPDATE_TEST(state, value) {
            state.test = { ...value }
        },
        INCREASE_COUNT(state, _id) {
            // console.log(_id)
            let isTheTestExists = state.selectedTests.find(test => test._id === _id)
            // console.log(isTheTestExists)

            if (!isTheTestExists) {
                state.count++;
                // localStorage.setItem('seletedTests',)

                const test = state.tests.find(test => test._id === _id)
                // console.log('test', test)
                state.selectedTests.push(test)
                // const localTests = state.selectedTests.map(test => test)
                // console.log(localTests);
                localStorage.setItem('count', state.count)
                localStorage.setItem('selectedTests', JSON.stringify(state.selectedTests))
                Vue.$vToastify.success(`${test.testname} ärnu placerad i kundvageen`)

            } else {
                return
            }
        },
        SHOW_CART_COMPONENT(state) {
            console.log('show action')
            state.showSelectedTests = true;
        },
        CLOSE_CART_COMPONENT(state) {
            state.showSelectedTests = false;
        },
        TOTAL_AMOUNT(state, total) {
            state.totalAmount = total
            localStorage.setItem('total_amount', state.totalAmount)
        },
        DELETE_TEST_CART(state, id) {
            console.log('move to state')
            const testInfo = state.selectedTests.find((test) => test._id === id)
            const remainingTests = state.selectedTests.filter((test) => test._id !== id)
            state.selectedTests = remainingTests
            state.count--
            localStorage.setItem('selectedTests', JSON.stringify(state.selectedTests))
            localStorage.setItem('count', state.count)
            console.log(state.selectedTests)
            Vue.$vToastify.success(`${testInfo.testname} is removed from the cart`)
        },
        DELETE_SELECTED_TESTS(state) {
            console.log('delete selected tests')
            state.selectedTests = [],
                state.count = 0;
            Vue.$vToastify.success(` Produkten är borttagen från kundvagen `)

        },
        UPDATE_PAGINATED_TESTS(state, value) {
            state.paginatedTests = value
        },
        UPDATE_PAGINATED_PAGES(state, value) {
            state.numberOfPages = value
        },
        INCREASE_QUANTITY(state, id) {
            const findTest = state.selectedTests.find(test => test._id === id)
            // console.log(findTest)
            findTest.quantity++
            state.tests.filter(test => test._id === id).push(findTest)
        },
        DECREASE_QUANTITY(state, id) {
            const findTest = state.selectedTests.find(test => test._id === id)
            // console.log(findTest)
            findTest.quantity--
            console.log(findTest.quantity)
            if (findTest.quantity < 1) {
                state.tests.filter(test => test._id === id)
            }
            state.tests.filter(test => test._id === id).push(findTest)
        },


    },
    namespaced: true

}