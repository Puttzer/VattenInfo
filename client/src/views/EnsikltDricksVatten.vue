<template>
  <v-main class="container">
    <v-flex>
      <v-row> </v-row>
      <v-row d-flex justify="center">
        <h1>Enskilt dricksvatten</h1>
      </v-row>
      <v-row class="d-flex justify-center font-weight-light blue--text mt-6">
        <h3>Analyser för dig med enskild brunn</h3>
      </v-row>
      <v-row class="d-flex justify-center packet-height">
        <v-card
          v-for="(test, index) in this.filterPaketTest"
          :key="index"
          width="250px"
          height="300px"
          class="ma-4"
        >
          <v-list
            class="d-flex flex-column justify-center ma-0 pa-0"
            id="testList"
          >
            <v-img
              id="testImage"
              :src="`http://localhost:4000/${test.image}`"
              height="150px"
              width="300px"
              name="testimage"
              class="ma-0 pa-0"
            ></v-img>
            <v-divider></v-divider>
            <div class="ma-2">
              <h2 class="Heading-2" id="testName">
                {{ test.testname }}
              </h2>
              <p id="testCategory">
                <strong>Kategori : </strong> {{ test.category }}
              </p>
              <!-- <p id="testType">
                <strong>Test type : </strong>{{ test.testtype }}
              </p>

              <p id="testLong">Detailjer :{{ test.description }}</p> -->
              <p id="testPrice">Pris :{{ test.price }} SEK</p>
            </div>
            <div class="d-flex flex-row">
              <v-btn
                class="btnColor white--text ma-1"
                @click="moveToIndividual(test._id, test)"
                >Läs mer</v-btn
              >
              <v-btn
                class="btnColor white--text ma-1"
                @click="moveToCart(test._id)"
                >köp</v-btn
              >
            </div>
          </v-list>
        </v-card>
      </v-row>
      <v-divider></v-divider>
      <v-row class="blue--text d-flex justify-center my-4">
        <h3>Individuell tester styckvis 1</h3>
      </v-row>
      <v-row>
        <v-card
          v-for="(test, index) in this.filterIndividuelStyckvis1"
          :key="index"
          width="200px"
          height="200px"
          class="ma-4 amber darken-1"
        >
          <v-list
            class="d-flex flex-column justify-center amber darken-1"
            id="testList ma-0 pa-0"
          >
            <!-- <v-img
              id="testImage"
              :src="`http://localhost:4000/${test.image}`"
              height="150px"
              width="300px"
              name="testimage"
              class="ma-0 pa-0"
            ></v-img>
            <v-divider></v-divider> -->
            <div class="ma-2">
              <h2 class="Heading-2" id="testName">
                {{ test.testname }}
              </h2>
              <p id="testCategory">
                <strong>Kategori : </strong> {{ test.category }}
              </p>
              <!-- <p id="testType">
                <strong>Test type : </strong>{{ test.testtype }}
              </p>

              <p id="testLong">Detailjer :{{ test.description }}</p> -->
              <p id="testPrice">Pris :{{ test.price }} SEK</p>
            </div>
            <div class="d-flex justify-center">
              <!-- <v-btn
                class="btnColor white--text ma-1"
                @click="moveToIndividual(test._id, test)"
                >Läs mer</v-btn
              > -->
              <v-btn
                class="btnColor white--text ma-1"
                @click="moveToCart(test._id)"
                >köp</v-btn
              >
            </div>
          </v-list>
        </v-card>
      </v-row>
      <v-divider></v-divider>
      <v-row class="blue--text d-flex justify-center my-4">
        <h3>Individuell tester styckvis 2</h3>
      </v-row>
      <v-row>
        <v-card
          v-for="(test, index) in this.filterIndividuelStyckvis2"
          :key="index"
          width="200px"
          height="200px"
          class="ma-4 green darken-1"
        >
          <v-list
            class="d-flex flex-column justify-center green darken-1"
            id="testList ma-0 pa-0"
          >
            <!-- <v-img
              id="testImage"
              :src="`http://localhost:4000/${test.image}`"
              height="150px"
              width="300px"
              name="testimage"
              class="ma-0 pa-0"
            ></v-img>
            <v-divider></v-divider> -->
            <div class="ma-2 white--text">
              <h2 class="Heading-2" id="testName">
                {{ test.testname }}
              </h2>
              <p id="testCategory">
                <strong>Kategori : </strong> {{ test.category }}
              </p>
              <!-- <p id="testType">
                <strong>Test type : </strong>{{ test.testtype }}
              </p>

              <p id="testLong">Detailjer :{{ test.description }}</p> -->
              <p id="testPrice">Pris :{{ test.price }} SEK</p>
            </div>
            <div class="d-flex justify-center">
              <!-- <v-btn
                class="btnColor white--text ma-1"
                @click="moveToIndividual(test._id, test)"
                >Läs mer</v-btn
              > -->
              <v-btn
                class="btnColor white--text ma-1"
                @click="moveToCart(test._id)"
                >köp</v-btn
              >
            </div>
          </v-list>
        </v-card>
      </v-row>
    </v-flex>
  </v-main>
</template>

<script>
import { mapState } from "vuex";
export default {
  name: "EnsikltDricksVatten",
  data() {
    return {};
  },
  mounted() {
    this.$store.dispatch("tests/getTests");
    console.log(this.$root);
  },
  computed: {
    ...mapState(["tests"]),
    filterPaketTest() {
      return this.tests.tests.filter(
        (test) =>
          test.testtype === "Packet pris" &&
          test.category === "Enskilt dricksvatten"
      );
    },
    filterIndividuelStyckvis1() {
      return this.tests.tests.filter(
        (test) =>
          test.testtype === "Analyser Styckvis 1" &&
          test.category === "Enskilt dricksvatten"
      );
    },
    filterIndividuelStyckvis2() {
      return this.tests.tests.filter(
        (test) =>
          test.testtype === "Analyser styckvis 2" &&
          test.category === "Enskilt dricksvatten"
      );
    },
  },
  methods: {
    moveToCart(id) {
      console.log("move to cart", id);
      console.log(this);
    },
    moveToIndividual(id, test) {
      console.log(id);
      this.$router.push({
        name: "TestStartsidan",
        params: { id: id, test },
      });
    },
  },
};
</script>

<style scoped>
.container {
  width: 80vw;
  margin-top: 20px;
  display: flex;
  justify-content: center;
}
.packet-height {
  min-height: 400px;
}
p {
  margin: 0;
  padding: 0;
}
</style>