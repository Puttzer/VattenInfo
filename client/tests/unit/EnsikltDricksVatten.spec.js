import Vue from 'vue'
import Vuetify from 'vuetify'

// Components
import EnsikltDricksVatten from '../../src/views/EnsikltDricksVatten'

// Utilities
import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vuex from 'vuex'
import tests from '../../src/store/modules/tests'

const localVue = createLocalVue()
localVue.use(Vuex)


describe('EnsikltDricksVatten.vue', () => {
    let vuetify
    let actions, state, store, mutations

    beforeEach(() => {
        vuetify = new Vuetify()
        state = {
            tests: [],

        }

        store = new Vuex.Store({
            modules: {
                tests: {
                    state,
                    actions,
                    mutations,
                    namespaced: true
                }
            }
        })
    })


    it('should have a custom title and match snapshot', () => {
        const wrapper = shallowMount(EnsikltDricksVatten, {
            localVue,
            vuetify,
            store
        })

        // With jest we can create snapshot files of the HTML output
        // expect(wrapper.html()).toMatchSnapshot()

        // We could also verify this differently
        // by checking the text content
        const title = wrapper.find('h2.ensiklt-heading')
        const resultTitle = 'Analyser i lokalt laboratorium';

        expect(title.text()).toBe(resultTitle)
    })

    it('should have a title with h3 element', () => {
        const wrapper = shallowMount(EnsikltDricksVatten, {
            localVue,
            vuetify,
            store
        })

        // With jest we can create snapshot files of the HTML output
        // expect(wrapper.html()).toMatchSnapshot()

        // We could also verify this differently
        // by checking the text content
        const title = wrapper.find('h3.h3-heading')
        const resultTitle = 'Analyser styckvis, prisklass 2 250 kr/st';

        expect(title.text()).toBe(resultTitle)
    })
    // it('should emit an event when the action v-btn is clicked', () => {
    //     const wrapper = mount(CustomCard, {
    //         localVue,
    //         vuetify,
    //         propsData: { title: 'Foobar' },
    //     })

    //     const event = jest.fn()
    //     const button = wrapper.find('.v-btn')

    //     // Here we bind a listener to the wrapper
    //     // instance to catch our custom event
    //     // https://vuejs.org/v2/api/#Instance-Methods-Events
    //     wrapper.vm.$on('action-btn:clicked', event)

    //     expect(event).toHaveBeenCalledTimes(0)

    //     // Simulate a click on the button
    //     button.trigger('click')

    //     // Ensure that our mock event was called
    //     expect(event).toHaveBeenCalledTimes(1)
    // })

})
