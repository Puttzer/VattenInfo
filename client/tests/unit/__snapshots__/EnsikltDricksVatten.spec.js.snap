// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnsikltDricksVatten.vue should have a custom title and match snapshot 1`] = `
<v-main class="container">
  <v-flex>
    <v-row d-flex="" justify="center">
      <h2 class="ensiklt-heading">
        Analyser för dig med egen brunn eller gemensam brunn för upp till 50
        personer
      </h2>
    </v-row>
    <v-row class="d-flex justify-center font-weight-bold blue--text mt-6">
      <v-col cols="11" class="d-flex justify-center">
        <p class="bread">
          Här hittar du analyser för dig med enskilt vatten (vatten för privat
          dricksvattenförsörjning för färre än 50 personer eller med ett flöde
          på mindre än 10 kubikmeter).<br>
          Om du har anlagt en ny brunn så rekommenderas att prova vatten efter
          någon månads användning. I andra fall rekommenderar Livsmedelsverket
          att du testar ditt eget vatten minst vart tredje år.<br>
          Är det fler än två fastigheter som är anslutna till samma brunn, om
          vattnet kommer från en sjö eller passerar en avsaltningsanläggning
          bör vattnet testas minst en gång per år.<br> <span> OBS! </span> Vatten som försörjer fler än 50 personer eller
          har ett flöde på minst 10 m3 är dricksvatten enligt
          Livsmedelslagstiftningen. Klicka här för att komma till
          ackrediterade analyser för registrerade anläggningar.
        </p>
      </v-col>
    </v-row>
    <v-row class="d-flex justify-center mt-6">
      <v-col cols="11" class="d-flex flex-column justify-center">
        <h2 class="d-flex justify-center text--black">
          Analyser i lokalt laboratorium
        </h2>
        <p class="bread font-weight-light blue--text">
          Det lokala laboratoriet analyserar enligt akrediterade metoder, men
          är inte ackrediterat. Därför är det lämpligt för enskilt vatten.
          Vårt lokala laboratorie erbjuder snabbare svarstider och
          direktkoppling till rådgivning.
        </p>
      </v-col>
    </v-row>
    <v-row class="d-flex justify-center packet-height"></v-row>
    <v-divider></v-divider>
    <v-col cols="12" class="d-flex justify-center flex-row">
      <div>
        <v-row class="blue--text d-flex justify-center">
          <h3>Analyser styckvis, prisklass 1 <br>200 kr/st</h3>
        </v-row>
        <v-col cols="12" class="d-flex flex-column justify-center"></v-col>
      </div>
      <div>
        <v-row class="blue--text d-flex justify-center my-4">
          <h3>Analyser styckvis, prisklass 2 <br>250 kr/st</h3>
        </v-row>
        <v-col cols="" class="d-flex flex-column justify-center"></v-col>
      </div>
    </v-col>
    <v-divider></v-divider>
    <v-row class="d-flex justify-center mt-6">
      <v-col cols="10">
        <h1 class="d-flex justify-center text--black">
          Ackrediterade analyser för enskilt dricksvatten
        </h1>
        <h2 class="bread font-weight-light blue--text">
          Analyser för dig som önskar få en bedömning från ett ackrediterat
          laboratorium eller analysera specifika parametrar så som metaller,
          samt räkna bakterier.
        </h2>
      </v-col>
    </v-row>
    <v-col class="d-flex justify-center packet-height"></v-col>
    <v-divider></v-divider>
    <v-row class="blue--text d-flex justify-center my-4">
      <h3>Ackrediterade analyser styckvis</h3>
    </v-row>
    <v-col class="d-flex flex-column justify-center"></v-col>
    <v-col class="d-flex align-center justify-center">
      <v-btn icon="" large="">
        <v-icon>arrow_upward</v-icon>
      </v-btn>
    </v-col>
  </v-flex>
</v-main>
`;
